### PR TITLE
Setup

### PR DESCRIPTION
**Pull request by vigehi**
Hello reviewer, 

According to the project requirements i hve tried my best to

1. Create a method to decode a Morse code character, takes a string parameter, and return the corresponding character in uppercase (e.g. decode_char(".-") returns "A").
2. Create a method to decode an entire word in Morse code, takes a string parameter, and return the string representation. Every character in a word will be separated by a single space (e.g. decode_word("-- -.--") returns "MY") 
3. Create a method to decode the entire message in Morse code, takes a string parameter, and return the string representation. Every word will be separated by 3 spaces (e.g.

      decode("-- -.--   -. .- -- .")

    returns "MY NAME").

    Reuse methods whenever possible. Make sure that your code is DRY.

    Once your code is ready use it to decode the message from the old bottle:

      .-   -... --- -..-   ..-. ..- .-.. .-..   --- ..-.   .-. ..- -... .. . ...
  It is a pair programming task but i dont have a partner

Thank you for taking your time to review mu code

### CODE REVIEW

File: morsecode.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    'v' => '...-',
+    'w' => '.--',
+    'x' => '-..-',
+    'y' => '-.--',
+    'z' => '--..',
+    ' ' => ' '
+  }
+
+  check.key(char).upcase
+end
```
Comment (AmaduKamara) [inline morsecode.rb:35]: - [x] When I run the other tests, they all pass but the one to print "MY NAME" does not seem to be working. It throws an arrow. See the screenshot below. Kindly fix the error it's returning 👍 so everything works fine as expected.

image

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️♻️♻️ 

Hi @vigehi,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✅ You used Gitflow correctly 👍 
✅ Linter checks passed 👍 
✅ Good descriptive PR 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: morsecode.rb (hunk 1 above)
Comment (vigehi) [reply morsecode.rb:35]: Thank you for the review

**Review by Nemwel-Boniface — APPROVED** [review]
## Hello @vigehi    :wave: ,
I am Nemwel your reviewer for this PR :man_beard:  

Thank you for meeting all the project requirements good job! :partying_face: 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! :partying_face: 🎉 



## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️ 
Met all the milestone requirements ✔️ 

## [Optional] suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**